controllers/authorizeService.js

```diff
@@ -0,0 +1,113 @@
+     config = require('../config'),
+     uuid = require('node-uuid');
+
+var authorizeService = (function () {
+
+    /**
+     * Generates am aìKey.
+     */
+    var generateApiKey = function () {
+        var apiKey = uuid.v4();
+
+        return apiKey;
+    };
+
+    /**
+     * Check if the remote cliente is the WStore; otherwise return an error.
+     */
+    var checkRemoteClient = function (ip) {
+        var storeHostname = config.appHost;
+        var remoteHostname = ip.replace(/^.*:/, ''); // Parse IPv4 embedded in IPv6
+
+        return remoteHostname === storeHostname;
+    }
+
+    /**
+     * Generates and send an apiKey for the url service specifed in the request body. The apiKey is saved in "uncommitted" state.
+     *
+     * @param  {Object} req     Incoming request.
+     * @param  {Object} res     Outgoing object.
+     */    
+    var getApiKey = function (req, res) {
+        // Check if request is from WStore
+        var fromWStore = checkRemoteClient(req.ip);
+
+        if (!fromWStore) {
+            res.status(401).json({error: 'Invalid remote client'});
+
+        } else {
+
+            try{
+
+                // Check the request and extract the url
+                var url = JSON.parse(req.body).url;
+                
+                if (url) {
+
+                    // Generate and save apiKey
+                    var apiKey = generateApiKey();
+                    var service = new AccountingService();
+                    service.url = url;
+                    service.apiKey = apiKey;
+                    service.state = 'UNCOMMITTED';
+
+                    service.save(function (err) {
+
+                        if (err) {
+                            res.status(500).send();
+
+                        } else {
+
+                            res.status(201).json({apiKey: apiKey});
+                        }
+                    });
+
+                } else {
+                    res.status(400).json({error: 'Url missing'});
+                }
+
+            } catch (e) {
+                res.status(400).json({ error: 'Invalid body' });
+            }
+        }
+    };
+
+    /**
+     * Change the apiKey state to "commited".
+     *
+     * @param  {Object} req      Incoming request.
+     * @param  {Object} res      Outgoing response.
+     */
+    var commitApiKey = function (req, res) {
+        // Check if request is from WStore
+        var fromWStore = checkRemoteClient(req.ip);
+
+        if (!fromWStore) {
+            res.status(401).json({error: 'Invalid remote client'});
+
+        } else {
+
+            // Update the apiKey state
+            var apiKey = req.params.apiKey;
+
+            AccountingService.update({apiKey: apiKey}, { $set: {state: 'COMMITTED'}}, function (err, rawResp) {
+                if (err) {
+                    res.status(500).send();
+                } else if (rawResp.nModified !== 1) {
+                    res.status(400).json({error: 'Invalid API Key'});
```

Comment:
This is not a 400... 400 is only returned when the body is invalid (invalid JSON...). In this case you must return 404 since the apiKey comes from the URL and it does not exist.